### PR TITLE
Migrate GiftedChat to TypeScript

### PR DESCRIPTION
**Pull request by xcarpentier**
Migrate the project to TypeScript.

cc @slorber @EvanBacon  ;)

### CODE REVIEW

**Comment by EvanBacon** [comment]
thanks for this

**Comment by xcarpentier** [comment]
Hi @EvanBacon! What do you think if we change a bit the examples files structure and make a single app example by merging others?

And also do something like @slorber did every time when he makes an RN lib.
I mean an expo app at the root and an npm script before publish. It's really more easy for dev cycle of a lib! see it here:
- https://github.com/slorber/react-native-animation-hooks
- I copied his awesome idea too in my last packages: https://github.com/xcarpentier/react-native-suggester or https://github.com/xcarpentier/rn-verifcode

**Comment by EvanBacon** [comment]
@xcarpentier yeah, I think one example would be best. It'd also be great to make things like react-native-video optional now that Expo (specifically expo-av) has web support.

**Comment by xcarpentier** [comment]
OK, will do it in another PR.

@EvanBacon can you precise your suggestion about web support (ie. expo-av)?

File: src/MessageContainer.js [hunk 1]
```diff
@@ -1,253 +0,0 @@
...
-
-  render() {
-    if (this.props.messages.length === 0) {
-      return <View style={styles.container} />;
-    }
-    return (
-      <View style={this.props.alignTop ? styles.containerAlignTop : styles.container}>
-        {this.state.showScrollBottom && this.props.scrollToBottom ? this.renderScrollToBottomWrapper() : null}
-        <FlatList
-          ref={(ref) => (this.flatListRef = ref)}
```
Comment (noelsp) [inline src/MessageContainer.js]: It was good!

File: src/MessageContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+      <View
+        style={
+          this.props.alignTop ? styles.containerAlignTop : styles.container
+        }
+      >
+        {this.state.showScrollBottom && this.props.scrollToBottom
+          ? this.renderScrollToBottomWrapper()
+          : null}
+        <FlatList
+          ref={this.flatListRef}
```
Comment (noelsp) [inline src/MessageContainer.tsx]: Here is an error. This means that when sending a new message, do not scroll down. I am right?

Comment (xcarpentier) [reply src/MessageContainer.tsx]: I you have time make PR, it will be more explicit. Thanks

**Pull request merged**